api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/support/oauth/OAuthProperties.java

```diff
@@ -36,6 +36,12 @@
      */
     private boolean replicateSessions;
 
+    /**
+     * Control the CSRF cookie settings in OAUTH authentication flows.
+     */
+    @NestedConfigurationProperty
+    private CsrfCookieProperties csrfCookie = new CsrfCookieProperties();
```

Comment:
By the way, please also add this section to the OAUTH docs, referencing `csrf-cookie` as the path for the property lookup.

```
{% include casproperties.html properties="cas.authn.oauth.xyz" %}
```